### ISSUE TITLE
Fix S2931 FN with fields of type IDisposable

### ISSUE THREAD

**Issue by RobertSimonsSMA** [SonarSource/sonar-dotnet#2957]
### Description

The Rule S2931: (Classes with "IDisposable" members should implement "IDisposable") ignores  fields of type IDisposable

### Repro steps

1. Enable Rule S2931
2. Write class with a private field of a type that implements IDisposable and assign it in the constructor.
3. Build -> S2931 is triggered
4. Change type of field to IDisposable 
```
public class NonDisposableClassWithDisposableMember
{
    private readonly IDisposable _disposable;

    public NonDisposableClassWithDisposableMember()
    {
        _disposable = new DisposableStuff();
    }

    private sealed class DisposableStuff : IDisposable
    {
        public void Dispose() { }
    }
}
```
5. Build -> S2931 is not triggered

### Expected behavior

In the last step, i expected the rule S2931 to be triggered.

### Actual behavior

In the last step, the rule S2931 was not triggered.

### Known workarounds

It is often possible to set a more concrete type for a field.

### Related information

* SonarC# Version: 7.11.0.8083
* Visual Studio Version: Visual Studio 2019 16.4.2
